sound/soc/sof/imx/imx-common.c

```diff
@@ -74,4 +74,79 @@ void imx8_dump(struct snd_sof_dev *sdev, u32 flags)
 }
 EXPORT_SYMBOL(imx8_dump);
 
+int imx8_parse_clocks(struct snd_sof_dev *sdev, struct imx_clocks *clks)
+{
+	int i;
+
+	clks->dsp_clks = devm_kcalloc(sdev->dev, clks->num_dsp_clks,
+				      sizeof(*clks->dsp_clks), GFP_KERNEL);
+	if (!clks->dsp_clks)
+		return -ENOMEM;
+
+	clks->dai_clks = devm_kcalloc(sdev->dev, clks->num_dai_clks,
+				      sizeof(*clks->dai_clks), GFP_KERNEL);
+	if (!clks->dai_clks)
+		return -ENOMEM;
+
+	for (i = 0; i < clks->num_dsp_clks; i++) {
+		clks->dsp_clks[i] = devm_clk_get(sdev->dev, clks->dsp_clks_names[i]);
+		if (clks->dsp_clks[i])
+			return PTR_ERR(clks->dsp_clks[i]);
+	}
+
+	for (i = 0; i < clks->num_dai_clks; i++)
+		clks->dai_clks[i] = devm_clk_get_optional(sdev->dev, clks->dai_clks_names[i]);
```

Comment:
If one of the optional clocks is null, is it ok to continue here?

Reply:
Yes. Some clocks are optional. Then calling enable/prepare on NULL clocks just ignores them.

e.g

```
int clk_enable(struct clk *clk)                                                                                                                                                                            
{
»       if (!clk)
»       »       return 0;

»       return clk_core_enable_lock(clk->core);
}
```